Library/Homebrew/cask/artifact/abstract_uninstall.rb

```diff
@@ -44,6 +44,8 @@ def initialize(cask, directives)
         directives[:signal] = Array(directives[:signal]).flatten.each_slice(2).to_a
         @directives = directives
 
+        # This is already included when loading from the API.
+        return if cask.loaded_from_api?
```

Comment:
I believe this was causing the `kext` caveat to show up twice when installing from the API. At the very least, it was showing up twice in the generated JSON.